File: src/Side.java
import javafx.scene.transform.Rotate;
import javafx.geometry.Point3D;

public enum Side {
  FRONT(Rotate.Z_AXIS),
  BACK(Rotate.Z_AXIS),
  LEFT(Rotate.X_AXIS),
  RIGHT(Rotate.X_AXIS),
  UP(Rotate.Y_AXIS),
  DOWN(Rotate.Y_AXIS),
  INSIDE(null);

  private Point3D value;

  private Side(Point3D value) {
    this.value = value;
  }

  public Point3D getValue() {
    return this.value;
  }
}

File: Assets/Scripts/Side.cs
using UnityEngine;

public enum Side
{
  FRONT,
  BACK,
  LEFT,
  RIGHT,
  UP,
  DOWN,
  INSIDE
}

public static class SideExtensions
{
  public static Vector3 GetNonAxis(this Side side) => side switch
  {
    Side.FRONT => Vector3.forward,
    Side.BACK => Vector3.forward,
    Side.LEFT => Vector3.right,
    Side.RIGHT => Vector3.right,
    Side.UP => Vector3.up,
    Side.DOWN => Vector3.up,
    Side.INSIDE => Vector3.zero,
    _ => Vector3.zero,
  };
}
